lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -381,58 +340,115 @@ class TextEditingElement {
+      event.preventDefault();
+      _onAction(_inputConfiguration.inputAction);
+    }
+  }
+}
+
+/// IOS/Safari behaviour for text editing.
+///
+/// In iOS, the virtual keyboard might shifts the screen up to make input
+/// visible depending on the location of the focused input element.
+///
+/// Due to this [initializeElementPosition] and [updateElementPosition]
+/// strategies are different.
+///
+/// [disableTextEditing] is also different since the [_positionInputElementTimer]
+/// also needs to be cleaned.
+///
+/// inputmodeAttribute needs to be set for mobile devices. Due to this
+/// [initializeTextEditing] is different.
+class IOSTextEditingStrategy extends DefaultTextEditingStrategy {
+  /// Timer that times when to set the location of the input text.
   ///
-  /// It will be located exactly in the same place with the editable widgets,
-  /// however it's contents and cursor will be invisible.
+  /// This is only used for iOS. In iOS, virtual keyboard shifts the screen.
+  /// There is no callback to know if the keyboard is up and how much the screen
+  /// has shifted. Therefore instead of listening to the shift and passing this
+  /// information to Flutter Framework, we are trying to stop the shift.
   ///
-  /// Users can interact with the element and use the functionalities of the
-  /// right-click menu. Such as copy,paste, cut, select, translate...
-  void applyAllStyles() {
-    _style?.applyToDomElement(domElement);
-    _positionElement();
-  }
+  /// In iOS, the virtual keyboard shifts the screen up if the focused input
+  /// element is under the keyboard or very close to the keyboard. Before the
+  /// focus is called we are positioning it offscreen. The location of the input
+  /// in iOS is set to correct place, 100ms after focus. We use this timer for
+  /// timing this delay.
+  Timer _positionInputElementTimer;
+  static const Duration _delayBeforePositioning =
+      const Duration(milliseconds: 100);
 
-  void _positionElement() {
-    if (_canPosition && _geometricInfo != null) {
-      _geometricInfo.applyToDomElement(domElement);
-    }
+  /// Whether or not the input element can be positioned at this point in time.
+  ///
+  /// This is currently only used in iOS. It's set to false before focusing the
+  /// input field, and set back to true after a short timer. We do this because
+  /// if the input field is positioned before focus, it could be pushed to an
+  /// incorrect position by the virtual keyboard.
+  ///
+  /// See:
+  ///
+  /// * [_delayBeforePositioning] which controls how long to wait before
+  ///   positioning the input field.
+  bool _canPosition = true;
+
+  @override
+  void initializeTextEditing(
+    InputConfiguration inputConfig, {
+    @required _OnChangeCallback onChange,
+    @required _OnActionCallback onAction,
+  }) {
+    super.initializeTextEditing(inputConfig,
+        onChange: onChange, onAction: onAction);
+    inputConfig.inputType.configureDomElement(domElement);
```

Comment:
`configureDomElement()` is a generic name. In the future, people could easily add non-mobile configuration and expect it to be called on desktop browsers too.

Maybe let's rename it to something more specific (e.g. `configureInputMode`)?

Or we could inline it:
```dart
super.initializeTextEditing(inputConfig, onChange: onChange, onAction: onAction);
final String inputmode = inputConfig.inputType.inputmodeAttribute;
if (inputmodeAttribute != null) {
  domElement.setAttribute('inputmode', inputmode);
}
```

Reply:
done. renamed (I try to use more code between IOSTextEditingStrategy and AndroidTextEditingStrategy)